pkg/cmd/roachtest/tests/backup.go

```diff
@@ -1103,6 +1104,368 @@ func registerBackup(r registry.Registry) {
+		t.Fatal(err)
+	}
+	if _, err := conn.Exec(`SET CLUSTER SETTING storage.mvcc.range_tombstones.enabled = 't'`); err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for ranges to upreplicate.
+	if err := WaitFor3XReplication(ctx, t, conn); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the orders table. It's about 16 GB across 8 files.
+	t.Status("creating table")
+	if _, err := conn.Exec(`CREATE DATABASE tpch`); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := conn.Exec(`USE tpch`); err != nil {
+		t.Fatal(err)
+	}
+	createStmt, err := readCreateTableFromFixture(
+		"gs://cockroach-fixtures/tpch-csv/schema/orders.sql?AUTH=implicit", conn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := conn.ExecContext(ctx, createStmt); err != nil {
+		t.Fatal(err)
+	}
+
+	// Set up some helpers.
+	waitForStatus := func(
+		jobID string,
+		expectStatus jobs.Status,
+		expectRunningStatus jobs.RunningStatus,
+		duration time.Duration,
+	) {
+		if err := retry.ForDuration(duration, func() error {
+			var status string
+			var payloadBytes, progressBytes []byte
+			err := conn.QueryRowContext(
+				ctx, `SELECT status, progress, payload FROM system.jobs WHERE id = $1`, jobID).
+				Scan(&status, &progressBytes, &payloadBytes)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if jobs.Status(status) == jobs.StatusFailed {
+				var payload jobspb.Payload
+				if err := protoutil.Unmarshal(payloadBytes, &payload); err == nil {
+					t.Fatalf("job failed: %s", payload.Error)
+				}
+				t.Fatalf("job failed")
+			}
+			if jobs.Status(status) != expectStatus {
+				return errors.Errorf("expected job status %s, but got %s", expectStatus, status)
+			}
+			if expectRunningStatus != "" {
+				var progress jobspb.Progress
+				if err = protoutil.Unmarshal(progressBytes, &progress); err != nil {
+					t.Fatal(err)
+				}
+				if jobs.RunningStatus(progress.RunningStatus) != expectRunningStatus {
+					return errors.Errorf("expected running status %s, but got %s",
+						expectRunningStatus, progress.RunningStatus)
+				}
+			}
+			return nil
+		}); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	now := func() string {
+		var now string
+		if err = conn.QueryRowContext(ctx, `SELECT NOW()`).Scan(&now); err != nil {
+			t.Fatal(err)
+		}
+		return now
+	}
+
+	// Import the odd-numbered files.
+	t.Status("importing odd-numbered files")
+	files := []string{
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.1?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.3?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.5?AUTH=implicit'`,
+		`'gs://cockroach-fixtures/tpch-csv/sf-100/orders.tbl.7?AUTH=implicit'`,
+	}
+	var jobID string
+	err = conn.QueryRowContext(ctx, fmt.Sprintf(
+		`IMPORT INTO orders CSV DATA (%s) WITH delimiter='|', detached`, strings.Join(files, ", ")),
+	).Scan(&jobID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	waitForStatus(jobID, jobs.StatusSucceeded, "", 20*time.Minute)
```

Comment:
what's the intention behind running the IMPORT (and the other bulk jobs below) with detatched and polling the jobs table? Also i think we're trying to deprecate `retry.Duration()`?

Reply:
hm, maybe bc you want to hard cap how long an import takes?

Reply:
>hm, maybe bc you want to hard cap how long an import takes?

There's that (which could also be done with contexts, but it'd be more verbose), but also the fact that we sometimes have to check for non-terminal statuses -- specifically for the import cancellation (which needs to wait for the pausepoint), and the table drop (which needs to wait for the "waiting for MVCC GC" running status). Seemed easier to just use the same pattern everywhere, although it isn't strictly necessary everywhere. Happy to change it if you prefer though.

Reply:
I see. I think this pattern is fine, but it might be better to use the more up to date retry package semantics, which sets the initialBackoff to 50 ms by default. `ForDuration` begins the retry at 1ns interval, and i worry that reading the jobs table too often may affect the job performance in this test.

Reply:
Ended up with your suggestion of not detaching, and replaced `ForDuration`.